src/v/cluster/data_migration_types.h

```diff
@@ -227,6 +253,8 @@ struct migration_metadata
           .id = id, .migration = copy_migration(migration), .state = state};
     }
 
+    std::optional<data_migrations::state> next_ntp_state() const;
```

Comment:
why is this method name referring an `ntp` ? This is the next state of whole migration

Reply:
no, this method returns the next state NTPs need worked into, nullpot otherwise

Reply:
so maybe we can call it `next_partition_state` ? or `next_replica_state` ? `ntp` is an identifier